cdap-common/src/main/java/co/cask/cdap/common/namespace/Namespaces.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.namespace;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.proto.Id;
+import org.apache.twill.filesystem.Location;
+import org.apache.twill.filesystem.LocationFactory;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Utilities for namespaces
+ */
+public class Namespaces {
+  private Namespaces() {
+  }
+
+  public static Location getNamespaceHomeLocation(LocationFactory locationFactory, CConfiguration cConf,
```

Comment:
Instead of having everyone to call this method, you can provide a method that creates a `LocationFactory` that operates inside a given namespace:

```java
static LocationFactory createNamespacedLocationFactory(LocationFactory locationFactory,
                                                       CConfiguration cConf, Id.Namespace namespace) {
  LocationFactory namespaceBaseFactory = LocationFactories.namespace(locationFactory,
                                                cConf.get(Constants.Namespace.NAMESPACES_DIR));
  return LocationFactories.namespace(namespaceBaseFactory, namespace.getId());
}